dspace-server-webapp/src/test/java/org/dspace/app/rest/RequestItemRepositoryIT.java

```diff
@@ -0,0 +1,368 @@
+                        hasJsonPath("$._links.self.href", not(is(emptyOrNullString())))
+                )))
+                .andReturn();
+
+        // Clean up the created request.
+        String content = mvcResult.getResponse().getContentAsString();
+        Map<String,Object> map = mapper.readValue(content, Map.class);
+        String requestToken = String.valueOf(map.get("token"));
+        RequestItem ri = requestItemService.findByToken(context, requestToken);
+        requestItemService.delete(context, ri);
+
+        context.restoreAuthSystemState();
+    }
+
+    /**
+     * Test of getDomainClass method, of class RequestItemRepository.
+     */
+    @Test
+    public void testGetDomainClass() {
+        System.out.println("getDomainClass");
+        RequestItemRepository instance = new RequestItemRepository();
+        Class instanceClass = instance.getDomainClass();
+        assertEquals("Wrong domain class", RequestItemRest.class, instanceClass);
+    }
+
+    /**
+     * Verify that Spring Security's CSRF protection is working as we expect.
+     * We must test this using a simple non-GET request, as CSRF Tokens are not
+     * validated in a GET request.
+     *
+     * @throws java.lang.Exception passed through.
+     */
+    @Test
+    public void testRefreshTokenWithInvalidCSRF()
+            throws Exception {
+        // Login via password to retrieve a valid token
+        String token = getAuthToken(eperson.getEmail(), password);
+
+        // Remove "Bearer " from that token, so that we are left with the token itself
+        token = token.replace("Bearer ", "");
+
+        // Save token to an Authorization cookie
+        Cookie[] cookies = new Cookie[1];
+        cookies[0] = new Cookie(AUTHORIZATION_COOKIE, token);
+
+        getClient().perform(post(URI_ROOT)
+                .with(csrf().useInvalidToken().asHeader())
+                .secure(true)
+                .cookie(cookies))
+                // Should return a 403 Forbidden, for an invalid CSRF token
+                .andExpect(status().isForbidden())
+                // Verify it includes our custom error reason (from DSpaceApiExceptionControllerAdvice)
+                .andExpect(status().reason(containsString("Invalid CSRF token")))
+                // And, a new/updated token should be returned (as both server-side cookie and header)
+                // This is handled by DSpaceAccessDeniedHandler
+                .andExpect(cookie().exists("DSPACE-XSRF-COOKIE"))
+                .andExpect(header().exists("DSPACE-XSRF-TOKEN"));
+
+        //Logout
+        getClient(token).perform(post("/api/authn/logout"))
+                        .andExpect(status().isNoContent());
+    }
+
+    /**
+     * Verify that Spring Security's CORS settings are working as we expect.
+     *
+     * @throws java.lang.Exception passed through.
+     */
+    @Test
+    public void testCannotReuseTokenFromUntrustedOrigin()
```

Comment:
Same here, rename the test.  It's not about Token reuse...it's verifying that an item request cannot be made from an untrusted origin.

Reply:
Done.